lib/instantlock/instantlock.js

```diff
@@ -0,0 +1,351 @@
+ * @property {string} txid
+ * @property {string} signature
+ */
+class InstantLock {
+  constructor(arg) {
+    if (arg instanceof InstantLock) {
+      return arg.copy();
+    }
+    const info = InstantLock._from(arg);
+
+    this.inputs = info.inputs;
+    this.txid = info.txid;
+    this.signature = info.signature;
+    this.validate();
+    return this;
+  }
+
+  static get ISLOCK_REQUESTID_PREFIX() {
+    return 'islock';
+  }
+
+  /**
+   * @param {Buffer|Object|string} arg - A Buffer, JSON string or Object
+   * @returns {Object} - An object representing instantlock data
+   * @throws {TypeError} - If the argument was not recognized
+   * @private
+   */
+  static _from(arg) {
+    let info = {};
+    if (BufferUtil.isBuffer(arg)) {
+      info = InstantLock._fromBufferReader(BufferReader(arg));
+    } else if (isObject(arg)) {
+      info = InstantLock._fromObject(arg);
+    } else if (isHexaString(arg)) {
+      info = InstantLock.fromHex(arg);
+    } else {
+      throw new TypeError('Unrecognized argument for InstantLock');
+    }
+    return info;
+  }
+
+  static _fromObject(data) {
+    $.checkArgument(data, 'data is required');
+    const txid = data.txid || data.txId;
+    const { signature } = data;
+
+    return {
+      inputs: data.inputs,
+      txid,
+      signature,
+    };
+  }
+
+  /**
+   * @param {BufferReader} br - InstantLock data
+   * @returns {Object} - An object representing the InstantLock data
+   * @private
+   */
+  static _fromBufferReader(br) {
+    const info = {};
+    const inputsCount = br.readVarintNum();
+    info.inputs = [];
+    for (let i = 0; i < inputsCount; i += 1) {
+      const outpointHash = br.readReverse(SHA256_HASH_SIZE).toString('hex');
+      const outpointIndex = br.readInt32LE();
+      const outpoint = { outpointHash, outpointIndex };
+      info.inputs.push(outpoint);
+    }
+    info.txid = br.readReverse(SHA256_HASH_SIZE).toString('hex');
+    info.signature = br.read(BLS_SIGNATURE_SIZE).toString('hex');
+    return info;
+  }
+
+  /**
+   * @param {BufferReader} br A buffer reader of the block
+   * @returns {InstantLock} - An instance of InstantLock
+   */
+  static fromBufferReader(br) {
+    $.checkArgument(br, 'br is required');
+    const data = InstantLock._fromBufferReader(br);
+    return new InstantLock(data);
+  }
+
+  /**
+   * Creates InstantLock from a hex string.
+   * @param {String} string - A hex string representation of the instantLock
+   * @return {InstantLock} - An instance of InstantLock
+   */
+  static fromString(string) {
+    return InstantLock.fromBuffer(Buffer.from(string, 'hex'));
+  }
+
+  /**
+   * Creates InstantLock from a hex string.
+   * @param {String} string - A hex string representation of the instantLock
+   * @return {InstantLock} - An instance of InstantLock
+   */
+  static fromHex(string) {
+    return InstantLock.fromBuffer(Buffer.from(string, 'hex'));
```

Comment:
`InstatLock.fromString`?

Reply:
we have it as an alias above on line 110

Reply:
Yeah, let's just reuse this method here, or vise-versa to not duplicate `Buffer.from(string, 'hex')` logic

Reply:
got it! Done.